rllib/core/models/configs.py

```diff
@@ -46,38 +55,106 @@ def checked_build(self, framework, **kwargs):
+    """
+
+    input_dims: Union[List[int], Tuple[int]] = None
+    always_check_shapes: bool = False
+
+    @abc.abstractmethod
+    def build(self, framework: str):
+        """Builds the model.
+
+        Args:
+            framework: The framework to use for building the model.
+        """
+        raise NotImplementedError
+
+    @property
+    def output_dims(self) -> Optional[Tuple[int]]:
+        """Read-only `output_dims` are inferred automatically from other settings."""
+        return None
+
+
 @ExperimentalAPI
 @dataclass
 class _MLPConfig(ModelConfig):
     """Generic configuration class for multi-layer-perceptron based Model classes.
 
+    `output_dims` is reached by either the provided `output_layer_dim` setting (int) OR
+    by the last entry of `hidden_layer_dims`. In the latter case, no special output
+    layer is added and all layers in the stack behave exactly the same. If
+    `output_layer_dim` is provided, users might also change this last layer's
+    activation (`output_layer_activation`) and its bias setting
+    (`output_layer_use_bias`).
+
     This is a private class as users should not configure their models directly
     through this class, but use one of the sub-classes, e.g. `MLPHeadConfig` or
     `MLPEncoderConfig`.
 
     Attributes:
         input_dims: A 1D tensor indicating the input dimension, e.g. `[32]`.
-        hidden_layer_dims: The sizes of the hidden layers. If an empty list, only a
-            single layer will be built of size `output_dims[0]`.
+        hidden_layer_dims: The sizes of the hidden layers. If an empty list,
+            `output_layer_dim` must be provided (int) and only a single layer will be
+            built.
+        hidden_layer_use_bias: Whether to use bias on all dense layers in the network
+            (excluding a possible separate output layer defined by `output_layer_dim`).
         hidden_layer_activation: The activation function to use after each layer (
             except for the output).
         hidden_layer_use_layernorm: Whether to insert a LayerNorm functionality
             in between each hidden layer's output and its activation.
-        output_dims: A 1D Tensor indicating the size of the output layer. This may be
+        output_layer_dim: An int indicating the size of the output layer. This may be
             set to `None` in case no extra output layer should be built and only the
-            layers specified by `hidden_layer_dims` will part of the network.
-        output_activation: The activation function to use for the output layer, if any.
-        use_bias: Whether to use bias on all dense layers in the network (including
-            a possible output layer).
+            layers specified by `hidden_layer_dims` will be part of the network.
+        output_layer_use_bias: Whether to use bias on the separate output layer, if any.
+        output_layer_activation: The activation function to use for the output layer,
+            if any.
```

Comment:
Can we add here that the default  for output is linear? Also that the default for the others is relu?
Just so that folks realize that defining an output layer instead of the last hidden layer is not the same just like that.

Reply:
Ah, yes, great catch! Fixed.